x-pack/plugins/security_solution/server/endpoint/services/actions/action_list.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Logger } from '@kbn/core/server';
+import type { ActionDetails, ActionListApiResponse } from '../../../../common/endpoint/types';
+import type { SecuritySolutionRequestHandlerContext } from '../../../types';
+
+import {
+  getActionsForAgents,
+  getActionResponses,
+  getTimeSortedActionDetails,
+} from '../../utils/action_list_helpers';
+
+import {
+  categorizeActionResults,
+  categorizeResponseResults,
+  getActionCompletionInfo,
+  mapToNormalizedActionRequest,
+} from './utils';
+
+interface GetActionListParam {
+  actionTypes?: string[];
+  context: SecuritySolutionRequestHandlerContext;
+  elasticAgentIds?: string[];
+  endDate?: string;
+  logger: Logger;
+  page?: number;
+  pageSize?: number;
+  startDate?: string;
+  userIds?: string[];
+}
+
+export const getActionList = async ({
+  actionTypes,
+  context,
+  elasticAgentIds,
+  endDate,
+  logger,
+  page: _page,
+  pageSize,
+  startDate,
+  userIds,
+}: GetActionListParam): Promise<ActionListApiResponse> => {
+  const size = Math.floor(pageSize ?? 50 / 2);
+  const page = _page ?? 1;
+  const from = page <= 1 ? 0 : page * size - size + 1;
```

Comment:
This feels like its not doing what you expect. I copied this code into a function and test it out, and I think you are a) getting the same record in multiple pages and b) returning back more items that requested. Here is what I did in the browser:

```javascript
window.f = (_page = 1, pageSize=10) => {
  const size = Math.floor(pageSize ?? 50 / 2);
  const page = _page ?? 1;
  const from = page <= 1 ? 0 : page * size - size + 1;
    
  return { from, to: (from + size) };
}

// Test: 
f(1, 10)  // === {from: 0, to: 10}
f(2, 10). // === {from: 11, to: 21}
f(3, 10)  // === {from: 21, to: 31}
f(4, 10). // === {from: 31, to: 41}
```

Reply:
Also - I don't think you want the default to be `50` for `pageSize` ES normal page size is `10` I think

Reply:
Yeah. I'll change this to 10. We used a 50 for the current log as there are no page numbers on the UI. The new UI does have page numbers

Reply:
> tting the same record in multiple pages and b) returning back more items that requested. Here is what I did in the browser:

The test shows that it fetching next 10 items on every increment of page. That is expected.